### PR TITLE
EMP-339:  CRM 4 subsection array display (Alternative Quotes)

### PR DESCRIPTION
**Pull request by muyinatech**
**Changes made:**
- Removed DisplayField type to make display config simpler
- Added value field to ConfigField
- Made fields optional for SubSection
- Added macro for CRM4 Alternative Quotes
- Modified customDisplay macro to include CRM4 Alternative Quotes
- Removed duplicated code from formField macro
- Created formSubsection macro to remove duplicated code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Equiniti historical data frontend
 [![repo standards badge](https://img.shields.io/badge/endpoint.svg?&style=flat&logo=github&url=https%3A%2F%2Foperations-engineering-reports.cloud-platform.service.justice.gov.uk%2Fapi%2Fv1%2Fcompliant_public_repositories%2Fhmpps-template-typescript)](https://operations-engineering-reports.cloud-platform.service.justice.gov.uk/public-github-repositories.html#hmpps-template-typescript "Link to report")
-[![CircleCI](https://circleci.com/gh/ministryofjustice/hmpps-template-typescript/tree/main.svg?style=svg)](https://app.circleci.com/pipelines/github/ministryofjustice/laa-crime-equinity-historical-data-frontend)
+[![CircleCI](https://circleci.com/gh/ministryofjustice/laa-crime-equinity-historical-data-frontend/tree/main.svg?style=svg)](https://app.circleci.com/pipelines/github/ministryofjustice/laa-crime-equinity-historical-data-frontend)
```
Comment (muyinatech) [inline README.md:3]: Corrected - to point to our repo

File: server/services/crmDisplayService.ts [hunk 2]
```diff
@@ -92,22 +91,23 @@ export default class CrmDisplayService {
...
-        if (isConfigField(field)) {
-          // create display field using config & api field value
-          const displayField: DisplayField = {
-            label: field.label,
-            value: this.getApiFieldValue(crmResponse, field.apiField),
-            type: field.type,
+    if (fields) {
+      return fields
+        .map(field => {
+          if (isConfigField(field)) {
```
Comment (muyinatech) [inline server/services/crmDisplayService.ts:97]: This has been simplified by removing `DisplayField` type and now just populating `ConfigField.value` with the api field value.

File: server/views/components/formField.njk [hunk 3]
```diff
@@ -56,17 +23,21 @@
...
-                        {{ field.value }}
-                    {% endif %}
+                    {{ formFieldByType(field) }}
                 </dd>
             </div>
         </div>
     {% endif %}
 {% endmacro %}
+
+{% macro formFieldByType(field) %}
```
Comment (muyinatech) [inline server/views/components/formField.njk:33]: Created inner macro to reduce duplicated code.

File: server/views/components/navigationBar/macro.njk
Comment (muyinatech) [inline server/views/components/navigationBar/macro.njk:1]: Deleted - not used anywhere

File: server/views/components/navigationBar/template.njk
Comment (muyinatech) [inline server/views/components/navigationBar/template.njk:1]: Deleted - not used anywhere

File: server/services/crmDisplayService.test.ts [hunk 4]
```diff
@@ -210,28 +211,34 @@ describe('CRM Display Service', () => {
   })
 
   describe('getCrmSection()', () => {
+    const generalInformationSection: Section = {
+      sectionId: 'general-information',
+      subsections: [
+        {
+          fields: [
+            {
+              label: 'Has a previous application for an extension been made?',
```
Comment (gpjustice) [inline server/services/crmDisplayService.test.ts:220]: Just a suggestion - `Have you previously applied for an extension?`

File: server/services/crmDisplayService.test.ts [hunk 5]
```diff
@@ -303,30 +310,36 @@ describe('CRM Display Service', () => {
...
               {
                 label:
                   "Does your client or partner (if living with client as couple) get Income Support, Income Based Job Seeker's Allowance, Income Related Employment and Support Allowance or Guarantee State Pension Credit?",
                 value: 'Yes',
+                apiField: 'CapitalDetails.hasIncomeSupport',
               },
               {
                 label: 'How many dependants does your client have?',
                 value: 2,
+                apiField: 'CapitalDetails.numOfDependants',
```
Comment (gpjustice) [inline server/services/crmDisplayService.test.ts:324]: Suggestion -> `numberOfDependents`

**Review by gpjustice — APPROVED** [review]

File: server/services/crmDisplayService.test.ts (hunk 5 above)
Comment (muyinatech) [reply server/services/crmDisplayService.test.ts:324]: numOfDependants is correct and that is what is returned by the API.

**Pull request merged**